### ISSUE TITLE
Adding mat-sort-header breaks the templates flex box layout.

### ISSUE THREAD

**Issue by oleersoy** [angular/components#16952]
#### Reproduction

Use StackBlitz to reproduce your issue: https://stackblitz.com/fork/components-issue

Here's a stackblitz with the header column words centered using flexbox:
https://stackblitz.com/edit/angular-minimal-material-table-demo-center-header?file=src%2Fapp%2Fapp.component.html

Steps to reproduce:
1. Add mat-sort-header to the demo
 
 
#### Expected Behavior

The header should maintain the vertical layout of the words.

#### Actual Behavior
The words are flattened and joined without a space.

#### Environment

- Angular: 8.2.0
- CDK/Material: 8.1.3
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Ubuntu

**Comment by oleersoy** [comment]
Note that if we alter the directives CSS like this:

```
## Stackblitz Demo of Answer

https://stackblitz.com/edit/angular-minimal-material-table-demo-center-header?file=src%2Fapp%2Fapp.component.html

## If adding sorting

Be aware of this issue:

https://github.com/angular/components/issues/16952

Adding the sort directive adds this class:

    .mat-sort-header-button {
        border: none;
        background: 0 0;
        /* display: flex; */
        /* align-items: center; */
       padding: 0;
       cursor: inherit;
       outline: 0;
       font: inherit;
       color: currentColor;
}
```

Then that solves it.  As far as I can tell this is fine.  We still get the arrows indicating the sort direction and it looks fine.

**Comment by oleersoy** [comment]
I think the problem is that the sort buttons wraps the header content like this:

```
<button class="mat-sort-header-button" type="button" aria-label="Change sorting for annualDemandLevel"><div _ngcontent-djx-c1="">Annual</div><div _ngcontent-djx-c1="">Demand</div><div _ngcontent-djx-c1="">Level</div></button>

The button should really be a sibling of the header content or it should not change the flex layout of the header content.

**Comment by oleersoy** [comment]
Related SO posts:

https://stackoverflow.com/questions/57763295/altering-the-css-added-by-an-angular-directive

https://stackoverflow.com/questions/57750213/centering-text-vertically-in-a-mat-header-col

**Comment by andrewseguin** [comment]
This will likely be fixed by #13297 when we adjust the innerHTML of the sort header (removing the button)

**Comment by jelbourn** [comment]
Closing as probably fixed in #13297
